File: 1236-NThTribonacciNumber/1236-NThTribonacciNumber.java
// Last updated: 7/31/2025, 5:31:54 PM
class Solution {
    public int tribonacci(int n) {
        if(n==0){
            return 0;
        }
        if(n==1 || n==2){
            return 1;
        }
        int i=3;
        int n1=0, n2=1, n3=0, n4 =1;
        while(i<=n){
            n3 = n2+n1+n4;
            n1 = n2;
            n2 = n4;
            n4 = n3;
            i++;
            
        }
        return n3;
        
    }
}

File: 1236-NThTribonacciNumber/1236-NThTribonacciNumber.py
# Last updated: 7/31/2025, 5:31:59 PM
class Solution:
    def tribonacci(self, n: int) -> int:
        a=0
        b=1
        c=1
        if(n==0):
            return 0
        if(n==1 or n==2):
            return 1
        if(n>=3):
            i=3
            while(i<=n):
                s=a+b+c
                a=b
                b=c
                c=s
                i=i+1
        return s
            